client/register.go

```diff
@@ -57,6 +57,10 @@ func parseTimeout(val string) (time.Duration, error) {
 }
 
 func runRegister(cmd *Command, args []string) *ErrorStatus {
+	if _, ok := cli.Type().FieldByName("KeyFolder"); !ok {  
```

Comment:
Will get error UncachedHTTPClient is not defined

Reply:
sorry -- my suggestion was incomplete.

The error you are getting is because the type is defined in a separate package. If you import that, you can perform the proposed cast.

```
import (
....
	"github.com/pinterest/knox"
)
```
then:
```
	if _, ok := cli.(*knox.UncachedHTTPClient); !ok {  
```